### PR TITLE
added option where to branch from (if different than the default branch)

### PR DESCRIPTION
**Pull request by FitzgeraldKrudde**
I implemented specifying where to branch from.\
Useful when you want to branch from a different branch than the default branch.

I tested the change and verified the resulting PR in the Azure GUI. This looked OK to me.

Please review/QA/merge.

Also see: #49

### CODE REVIEW

File: src/doing/pr/create_pr.py [hunk 1]
```diff
@@ -129,6 +133,7 @@ def cmd_create_pr(
     command += f'--draft "{str(draft).lower()}" '
     command += f'--work-items "{work_item_id}" '
     command += f'--source-branch "{branch_name}" '
```
Comment (operte) [inline src/doing/pr/create_pr.py:135]: Shouldn't this also change? Also related to a previous question, about adding two options instead of just one.

**Review by operte — CHANGES_REQUESTED** [review]
I left a couple of comments. 

azdo allows picking the source and target branches. This PR assumes that these are always the same. I have two main questions:

- If we want to continue with this approach, I think we should pick a better option name.
- We could also allow the user to specify source and target branches separately, following the azdo cli more closely.

Not sure what the best option is. Let me know.

**Comment by FitzgeraldKrudde** [comment]
Our usecase is that we want to branch from (and to) a different branch than master, for example develop.
I.e. source and target branch name are intentionally combined.

I would prefer than a better named parameter as opposed to using 2 parameters.
I think starting-branch sounds like a good name then.

Is this OK?

**Comment by operte** [comment]
Yes, I see. I also believe it's not common to branch from one branch to a different one.

I thought a bit more about it and I'm even less happy with the other names, so let's keep `default-branch`. But let's change the docs a bit. I'll update my comments.

File: src/doing/pr/create_pr.py (hunk 1 above)
Comment (operte) [reply src/doing/pr/create_pr.py:135]: Nevermind, I got confused.

File: src/doing/pr/create_pr.py [hunk 2]
```diff
@@ -129,6 +133,7 @@ def cmd_create_pr(
...
+    command += f'--target-branch "{default_branch}" '
```
Comment (operte) [inline src/doing/pr/create_pr.py:136]: This sets the target branch, but I think we don't define where to branch from. I believe we need to change line 120 as well.

**Review by operte — CHANGES_REQUESTED** [review]

File: src/doing/pr/commands.py [hunk 3]
```diff
@@ -106,6 +106,15 @@ def close(pr_id):
     help="Open newly created issue in the web browser.",
     show_envvar=True,
 )
+@click.option(
+    "--default-branch",
+    "-d",
+    required=False,
+    default="",
+    type=str,
+    help="The name of the branch to branch from and to, only needed if different that the configured default branch",
```
[the remark this answers is not in the document]
Comment (FitzgeraldKrudde) [reply src/doing/pr/commands.py:115]: updated, thx!

File: src/doing/pr/commands.py [hunk 4]
```diff
@@ -106,6 +106,15 @@ def close(pr_id):
     help="Open newly created issue in the web browser.",
     show_envvar=True,
 )
+@click.option(
+    "--default-branch",
+    "-d",
```
[the remark this answers is not in the document]
Comment (FitzgeraldKrudde) [reply src/doing/pr/commands.py:111]: updated to "-b"

File: src/doing/pr/create_pr.py (hunk 2 above)
Comment (FitzgeraldKrudde) [reply src/doing/pr/create_pr.py:136]: The branch_name is the name of the new (to be created) branch with the prefix issueid in it.

File: src/doing/workon/commands.py [hunk 5]
```diff
@@ -78,6 +78,15 @@
     help="The number of story points to assign. Not assigned if not specified.",
     show_envvar=True,
 )
+@click.option(
+    "--default-branch",
+    "-d",
+    required=False,
+    default="",
+    type=str,
+    help="The name of the branch to branch from and to, only needed if different that the configured default branch",
```
Comment (operte) [inline src/doing/workon/commands.py:87]: Could you also update this line?

File: docs/reference/manual/pr_create.md [hunk 6]
```diff
@@ -15,6 +15,7 @@ doing pr create 1234 --draft -r "john.doe@company.com jane.doe@company.com"
 doing pr create 1234 --draft -r "@me jane.doe@company.com"
 doing pr create 1234 --draft --checkout
 doing pr create 1234 --delete-source-branch --self-approve --auto-complete
+doing pr create 1234 --default-branch develop
 ```
 
 !!! notes ""
```
Comment (operte) [inline docs/reference/manual/pr_create.md:21]: Line 25 should also be updated to let users know of this new option.

**Review by operte — CHANGES_REQUESTED** [review]

File: src/doing/workon/commands.py [hunk 7]
```diff
@@ -78,6 +78,15 @@
     help="The number of story points to assign. Not assigned if not specified.",
     show_envvar=True,
 )
+@click.option(
+    "--default-branch",
+    "-d",
```
[the remark this answers is not in the document]
Comment (FitzgeraldKrudde) [reply src/doing/workon/commands.py]: updated

File: docs/reference/manual/pr_create.md (hunk 6 above)
Comment (FitzgeraldKrudde) [reply docs/reference/manual/pr_create.md:21]: updated

File: src/doing/workon/commands.py (hunk 5 above)
Comment (FitzgeraldKrudde) [reply src/doing/workon/commands.py]: updated

File: src/doing/pr/create_pr.py (hunk 2 above)
Comment (FitzgeraldKrudde) [reply src/doing/pr/create_pr.py:136]: that is determined on lines 120/121

**Comment by FitzgeraldKrudde** [comment]
I am a bit of a python newbie.. I am not familiar with python unit testing

**Review by operte — APPROVED** [review]

**Pull request merged**